pkg/compact/downsample/downsample_test.go

```diff
@@ -562,11 +568,24 @@ func TestDownsampleAggrAndNonEmptyXORChunks(t *testing.T) {
 	mb := newMemBlock()
 	mb.addSeries(ser)
 
+	// Checking if able to iterate through the non-empty XOR chunks.
+	for _, c := range ser.chunks {
+		err := expandChunkIterator(c.Chunk.Iterator(reuseIt), &all)
+		testutil.Ok(t, err)
+	}
+
+	// Checking if XOR chunks get downsampled to 5m aggregated chunks.
+	aggregatedData := DownsampleRaw(all, 300_000)
+	for _, cn := range aggregatedData {
+		_, ok := cn.Chunk.(*AggrChunk)
+		testutil.Equals(t, true, ok, "Not able to convert XOR chunks to 5m donwsampled aggreagted chunks.")
+	}
```

Comment:
Can you please elaborate more about what you are testing here? I think what we want is to make sure after running `Downsample`, we will have all chunks becoming `AggrChunk` so the validation should be after running `Downsample`.

Reply:
@yeya24  elaboration:
There is same process being taken place in the `Downsample` where we first iterate through the XOR chunks and then downsample them using `DownsampleRaw` function passing in the desired resolution.
As these two processes are responsible for the functionality we are looking for, I wanted to be more specific while running test cases. Hence, chose to test the functions specifically like this.
As you can see in the test above at line 580 the operand on the right `cn.Chunk.(*AggrChunk)` would return `ok` as false, if we are not able to convert it to aggregated data, the `ok` would be returned as true if the chunks got successfully converted to aggr chunks

Reply:
I don't think this is the correct way to test the function itself, you are testing `expandChunkIterator` and `DownsampleRaw` now, not `Downsample`, that's different.

We don't have to run `expandChunkIterator` and `DownsampleRaw` separately unless you want to compare whether the result is expected or not.

In our case, try to find the newly created block after downsampling, then expand its chunks to make sure they are AggrChunk now.

Reply:
Ohh I get your point now, I'll get to it and make the changes!
Thanks for the review